### PR TITLE
[ECO-4650] fix: remove message grouping for Ably Batch calls

### PR DESCRIPTION
**Pull request by ttypic**
Resolves https://github.com/ably/kafka-connect-ably/issues/171
Jira issue [ECO-4650](https://ably.atlassian.net/browse/ECO-4650)

The kafka connector put every message sent to a single channel in one `BatchSpec`, meaning we treated them atomically and send them as a single `ProtocolMessage`.

This causes a problem when the total `ProtocolMessage` size gets way too big.

That's why now we put every message in its own `BatchSpec`

[ECO-4650]: https://ably.atlassian.net/browse/ECO-4650?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SimonWoolf** [comment]
@ttypic I assume you meant to tag me as a reviewer on this, not assign it to me?

**Comment by ttypic** [comment]
@SimonWoolf , oh, I am sorry 🙈 miss clicked

File: src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java [hunk 1]
```diff
@@ -113,20 +123,38 @@ private boolean isError(final HttpPaginatedResponse response) {
...
+                getAsJsonPrimitive("failureCount").getAsInt();
+
+            if (failureCount == 0) continue;
 
-            response = response.next();
-        } while (response != null);
+            JsonArray results = batchSpecResponse.getAsJsonObject().getAsJsonArray("results");
+
+            if (results == null || results.size() != 1)
+                throw AblyException.fromErrorInfo(new ErrorInfo("Inconsistent batch response: result field should contain single element", 500));
```
Comment (AndyTWF) [inline src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java:142]: Do we need an Ably error code here? We have 500, but should probably have 50000 too?

**Review by AndyTWF — APPROVED** [review]

**Review by SimonWoolf — COMMENTED** [review]
I'm not able to review the code in detail, will leave that for andy, but just looking at the tests it looks like like it successfully resolves the issue :+1:

**Pull request merged**